plugins/render/api_key.go

```diff
@@ -0,0 +1,72 @@
+package render
+
+import (
+	"context"
+
+	"github.com/1Password/shell-plugins/sdk"
+	"github.com/1Password/shell-plugins/sdk/importer"
+	"github.com/1Password/shell-plugins/sdk/provision"
+	"github.com/1Password/shell-plugins/sdk/schema"
+	"github.com/1Password/shell-plugins/sdk/schema/credname"
+	"github.com/1Password/shell-plugins/sdk/schema/fieldname"
+)
+
+type Config struct {
+	Profiles          map[string]Profile
+}
+
+type Profile struct {
+	APIKey        string `yaml:"apiKey"`
+}
+
+func APIKey() schema.CredentialType {
+	return schema.CredentialType{
+		Name:          credname.APIKey,
+		DocsURL:       sdk.URL("https://render.com/docs"),
+		ManagementURL: sdk.URL("https://dashboard.render.com/u/settings#api-keys"),
+		Fields: []schema.CredentialField{
+			{
+				Name:                fieldname.APIKey,
+				MarkdownDescription: "API Key used to authenticate to Render.",
+				Secret:              true,
+				Composition: &schema.ValueComposition{
+					Length: 30,
+					Charset: schema.Charset{
+						Uppercase: true,
+						Lowercase: true,
+						Digits:    true,
+					},
+				},
+			},
+		},
+		DefaultProvisioner: provision.EnvVars(defaultEnvVarMapping),
+		Importer: importer.TryAll(
+			importer.TryEnvVarPair(defaultEnvVarMapping),
+			TryRenderConfigFile(),
+		),
+	}
+}
+
+var defaultEnvVarMapping = map[string]sdk.FieldName{
+}
```

Comment:
> In this current state, the `render` CLI is not provisioned any credentials.
> 
> I've taken a look at it, and I have a few pointers, hope these help:
> 
>     * looks like when running `render config init`, the configuration gets saved in a file at `~/.render/config.yaml`, with this format:
> 
> 
> ```
> version: 1
> sshPreserveHosts: <the-value-you-select-during-setup>
> profiles:
>   default:
>     defaultRegion: <your-region>
>     apiKey: <your-api-key>
> ```
> 
>     * since there are no envvars that override any of those settings above, we would need to provision a file. Have a look at the `mysql` plugin for something similar:
> 
> 
> ```
> 		DefaultProvisioner: provision.TempFile(mysqlConfig, provision.Filename("my.cnf"), provision.AddArgs("--defaults-file={{ .Path }}")),
> ```
> 
>     * you would like to use the `AtFixedPath` helper function, to specify the location where this file would be (on MacOS, that is `~/.render/config.yaml`)
> 
> 
> Hope this helps. Please let us know if you have any questions, and do reach out in Discord or schedule a pair programming session, if you feel that could be useful.

I have made an update. I am not sure if what i did is correct.